### ISSUE TITLE
Warn about unnecessary blocks

### ISSUE THREAD

**Issue by crumblingstatue** [rust-lang/rust#91564]
rustc already warns about unnecessary semicolons:
```Rust
fn main() {
	;;;;;;;;;;;;
}
```
```
warning: unnecessary trailing semicolons
 --> mytest.rs:2:2
  |
2 |     ;;;;;;;;;;;;
  |     ^^^^^^^^^^^^ help: remove these semicolons
  ```
However, there is no warning about unnecessary blocks:
```Rust
fn main() {
	{}{}{}{}{}{}{}{}{}{}{}{}{}{}{}
}
```

Is this feasible to implement? Should this be the responsibility of another tool? (clippy, rustfmt)

**Comment by scottmcm** [comment]
Given that there's also a rustc lint about unnecessary parens, having things in rustc sounds reasonable to me.

(With the caveat that [braces have semantics](https://bluss.github.io/rust/fun/2015/10/11/stuff-the-identity-function-does/#rust-has-dedicated-syntax-for-this), so there are many cases where it probably shouldn't warn.  But empty blocks in general seem reasonable to warn about.)
